### PR TITLE
httpsinfo: Use Log4j 2.x

### PR DESCRIPTION
**Pull request by kingthorin**
- Update build file, CHANGELOG, and change code to use updated logging infrastructure.
    - Set minimum ZAP version to 2.10.
    - Change from slf4j12 v1.7.6 (which was behind as a dependency) to up-to-date log4j-slf4j-impl v2.14.1
    - Update CHANGELOG entry.

Part of: zaproxy/zaproxy#6376

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: addOns/httpsInfo/httpsInfo.gradle.kts [hunk 1]
```diff
@@ -13,7 +13,7 @@ zapAddOn {
 
 dependencies {
     implementation("com.github.spoofzu:DeepViolet:5.1.16")
-    implementation("org.slf4j:slf4j-log4j12:1.7.6") {
+    implementation("org.slf4j:slf4j-log4j12:1.7.30") {
```
Comment (thc202) [inline addOns/httpsInfo/httpsInfo.gradle.kts:16]: Should use the bridge for Log4j 2.

Comment (kingthorin) [reply addOns/httpsInfo/httpsInfo.gradle.kts:16]: Same/similar to OpenAPI?
```

    implementation("org.apache.logging.log4j:log4j-slf4j-impl:2.14.0") {
        // Provided by ZAP.
        exclude(group = "org.apache.logging.log4j")
    }

    testImplementation("org.apache.logging.log4j:log4j-core:2.14.0")
```

Comment (thc202) [reply addOns/httpsInfo/httpsInfo.gradle.kts:16]: Yes, but the `testImplementation` should not be needed.

Comment (kingthorin) [reply addOns/httpsInfo/httpsInfo.gradle.kts:16]: Thanks

Comment (thc202) [reply addOns/httpsInfo/httpsInfo.gradle.kts:16]: No problem.

**Comment by kingthorin** [comment]
Review addressed.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**